### PR TITLE
osutil: pid 1 should exit directly instead of trying to kill itself

### PR DESCRIPTION
**Pull request by xiang90**
/cc @crawford @marineam

### CODE REVIEW

File: pkg/osutil/interrupt_unix.go [hunk 1]
```diff
@@ -59,13 +59,18 @@ func HandleInterrupts() {
...
 
 		for _, h := range ihs {
 			h()
 		}
 		signal.Stop(notifier)
-		syscall.Kill(syscall.Getpid(), sig.(syscall.Signal))
+		pid := syscall.Getpid()
+		// exit directly if it is the "init" process, since the kernel will not help to kill pid 1.
+		if pid == 1 {
+			os.Exit(128 + int(sig.(syscall.Signal)))
```
Comment (crawford) [inline pkg/osutil/interrupt_unix.go]: Just exit with 0 in this case.

Comment (xiang90) [inline pkg/osutil/interrupt_unix.go]: ok

**Comment by xiang90** [comment]
@crawford Fixed. PTAL.

**Comment by crawford** [comment]
It occurred to me that you could also implement this by releasing interruptExitMu. Either way works.

LGTM

**Comment by xiang90** [comment]
@crawford Yea. I thought it for a minute and thought it would make tricky thing even more trickier by addressing it implicitly. But it does help to prevent the deadlock in potential bad cases anyway...

**Pull request merged**